sdv/cag/_utils.py

```diff
@@ -60,40 +61,6 @@ def _validate_table_name_if_defined(table_name):
         raise ValueError('`table_name` must be a string or None.')
 
 
-def _remove_columns_from_metadata(metadata, table_name, columns_to_drop):
-    """Remove columns from metadata, including column relationships.
-
-        Will raise an error if the primary key is being dropped.
-
-    Args:
-        metadata (dict, sdv.metadata.Metadata): The Metadata which contains
-            the columns to drop.
-
-        table_name (str): Name of the table in the metadata, where the column(s)
-            are located.
-
-        columns_to_drop (list[str]): The list of column names to drop from the
-            Metadata.
-
-    Returns:
-        (sdv.metadata.Metadata): The new Metadata, with the columns removed.
-    """
-    if isinstance(metadata, Metadata):
-        metadata = metadata.to_dict()
-    for column in columns_to_drop:
-        primary_key = metadata['tables'][table_name].get('primary_key')
-        if primary_key and primary_key == column:
-            raise ValueError('Cannot remove primary key from Metadata')
```

Comment:
do we still have to worry about the primary key being removed?

Reply:
I don't believe so. In CAGs, we remove columns that are one of the arguments of the CAG (ex - `high_column_name` for Inequality). Those user-defined columns have to pass the initial checks for the CAG and then only would they be removed.
A primary key has to be an ID or PII type. So there should not be overlap between user-defined columns in CAG and primary keys.

Reply:
That's a good point. Looking at the [predefined constraints list](https://docs.sdv.dev/sdv/concepts/constraints/predefined-constraints), I can see supporting these for primary keys. Indeed, there is an outstanding feature request for supporting constraints with primary keys:
- https://github.com/sdv-dev/SDV/issues/961

Therefore, I added the back the function `_remove_columns_from_metadata`.